.github/workflows/release.yml

```diff
@@ -0,0 +1,219 @@
+  IMAGE_NAME: ${{ github.repository_owner }}/trin
+  CARGO_TERM_COLOR: always
+
+jobs:
+  extract-version:
+    name: extract version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Extract version
+        run: echo "VERSION=$(echo ${GITHUB_REF#refs/tags/})" >> $GITHUB_OUTPUT
+        id: extract_version
+    outputs:
+      VERSION: ${{ steps.extract_version.outputs.VERSION }}
+
+  build:
+    name: build release
+    strategy:
+      matrix:
+        arch:
+          [
+            aarch64-unknown-linux-gnu,
+            x86_64-unknown-linux-gnu,
+            x86_64-apple-darwin,
+            aarch64-apple-darwin,
+            x86_64-pc-windows-gnu,
+          ]
+        include:
+          - arch: aarch64-unknown-linux-gnu
+            platform: ubuntu-20.04
+            profile: maxperf
+          - arch: x86_64-unknown-linux-gnu
+            platform: ubuntu-20.04
+            profile: maxperf
+          - arch: x86_64-apple-darwin
+            platform: macos-latest
+            profile: maxperf
+          - arch: aarch64-apple-darwin
+            platform: macos-latest
+            profile: maxperf
+          - arch: x86_64-pc-windows-gnu
+            platform: ubuntu-20.04
+            profile: maxperf
+
+    runs-on: ${{ matrix.platform }}
+    needs: extract-version
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+      - name: Get latest version of stable Rust
+        run: rustup update stable
+      - name: Install target
+        run: rustup target add ${{ matrix.arch }}
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+
+      # ==============================
+      # Apple M1 SDK setup
+      # ==============================
+
+      - name: Apple M1 setup
+        if: ${{ matrix.job.target == 'aarch64-apple-darwin' }}
+        run: |
+          echo "SDKROOT=$(xcrun -sdk macosx --show-sdk-path)" >> $GITHUB_ENV
+          echo "MACOSX_DEPLOYMENT_TARGET=$(xcrun -sdk macosx --show-sdk-platform-version)" >> $GITHUB_ENV
+
+      # ==============================
+      #       Builds
+      # ==============================
+
+      - name: Build trin for ${{ matrix.arch }}
+        run: |
+          cargo install cross
+          env PROFILE=${{ matrix.profile }} make build-${{ matrix.arch }}
+
+      - name: Move cross-compiled binary
+        if: matrix.arch != 'x86_64-pc-windows-gnu'
+        run: |
+          mkdir artifacts
+          mv target/${{ matrix.arch }}/${{ matrix.profile }}/trin ./artifacts
+
+      - name: Move cross-compiled binary (Windows)
+        if: matrix.arch == 'x86_64-pc-windows-gnu'
+        run: |
+          mkdir artifacts
+          mv target/${{ matrix.arch }}/${{ matrix.profile }}/trin.exe ./artifacts
+
+      # ==============================
+      #       Signing
+      # ==============================
+
+      - name: Configure GPG and create artifacts
+        env:
+          GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
```

Comment:
@carver do you have admin access to the repo, do you know if we have added `GPG_SIGNING_KEY` and `GPG_PASSPHRASE` as secrets?

Reply:
Looks like no secrets set:
![image](https://github.com/ethereum/trin/assets/205327/44420b34-2b3b-4033-a941-8e5d79840c85)

Reply:
I'm not sure what the standards are for this, but I don't think the answer is for someone to put their private gpg key & passphrase into the repo secrets. I guess we would just generate a new one. If you want to do that and send it to me over keybase, I can save it into those secrets under Actions.

Reply:
>  If you want to do that and send it to me over keybase, I can save it into those secrets under Actions.

Sent the key over keybase.

Reply:
Saved to the repo 👍🏻